logging.go

```diff
@@ -231,57 +261,133 @@ func Panicf(format string, a ...interface{}) {
 func Warningf(format string, a ...interface{}) {
 	printf(WarningLevel, format, a...)
 }
 
+// WarningStructured provides structured logging for log level >= warning.
+func WarningStructured(msg string, args ...interface{}) {
+	args = append(args, "level", "warning")
+	m := structuredMessage(msg, args...)
+	printp(WarningLevel, false, m)
+}
+
 // Infof prints logging if logging level >= info
 func Infof(format string, a ...interface{}) {
 	printf(InfoLevel, format, a...)
 }
 
+// InfoStructured provides structured logging for log level >= info.
+func InfoStructured(msg string, args ...interface{}) {
+	args = append(args, "level", "info")
+	m := structuredMessage(msg, args...)
+	printp(InfoLevel, false, m)
+}
+
 // Debugf prints logging if logging level >= debug
 func Debugf(format string, a ...interface{}) {
 	printf(DebugLevel, format, a...)
 }
 
+// DebugStructured provides structured logging for log level >= debug.
+func DebugStructured(msg string, args ...interface{}) {
+	args = append(args, "level", "debug")
+	m := structuredMessage(msg, args...)
+	printp(DebugLevel, false, m)
+}
+
 // Verbosef prints logging if logging level >= verbose
 func Verbosef(format string, a ...interface{}) {
 	printf(VerboseLevel, format, a...)
 }
 
-func doWrite(writer io.Writer, level Level, format string, a ...interface{}) {
-	fmt.Fprint(writer, prefixer.CreatePrefix(level))
+// VerboseStructured provides structured logging for log level >= verbose.
+func VerboseStructured(msg string, args ...interface{}) {
+	args = append(args, "level", "verbose")
+	m := structuredMessage(msg, args...)
+	printp(VerboseLevel, false, m)
+}
+
+// structuredMessage takes msg and an even list of args and returns a structured message.
+func structuredMessage(msg string, args ...interface{}) string {
+	if len(args)%2 != 0 {
+		return fmt.Sprintf(`msg=%s, logging_failure: "must provide an even number of arguments for structured logging"`,
+			msg)
+	}
+
+	var output strings.Builder
+	output.WriteString(fmt.Sprintf(`time="%s" msg="%s" `, time.Now().Format(defaultTimestampFormat), msg))
+
+	for i := 0; i < len(args)-1; i += 2 {
+		_, _ = output.WriteString(fmt.Sprintf(`%s="%s" `, args[i], args[i+1]))
+	}
+
+	return output.String()
+}
+
+// doWrite takes care of the low level writing to the output io.Writer.
+func doWrite(writer io.Writer, format string, a ...interface{}) {
 	fmt.Fprintf(writer, format, a...)
 	fmt.Fprintf(writer, "\n")
 }
 
+// printf prints log messages if they match the configured log level. A configured prefix is prepended to messages.
 func printf(level Level, format string, a ...interface{}) {
-	if logger.Filename == "" {
-		fmt.Fprint(os.Stderr, logFileReqFailMsg)
+	printp(level, true, format, a...)
+}
+
+// printp prints log messages if they match the configured log level. Messages are optionally prepended by a configured
+// prefix.
+func printp(level Level, printPrefix bool, format string, a ...interface{}) {
```

Comment:
nit: printWithPrefix ? 

that said, its in the same "spirit" as printf so im ok with keeping it as is as well.